tests/ignite/contrib/handlers/test_lr_finder.py

```diff
@@ -337,21 +337,41 @@ def test_lr_suggestion(lr_finder, to_save, dummy_engine, dataloader):
     assert 1e-4 <= lr_finder.lr_suggestion() <= 10
 
 
-def test_plot_single_param_group(
-    lr_finder, to_save_single_param_groups, dummy_engine_single_param_groups, dataloader_plot
+def test_apply_suggested_lr_single_param_groups(
+    lr_finder, mnist_to_save, dummy_engine_mnist, mnist_optimizer, mnist_dataloader
 ):
 
-    with lr_finder.attach(
-        dummy_engine_single_param_groups, to_save_single_param_groups, end_lr=20, smooth_f=0.04
-    ) as trainer_with_finder:
-        trainer_with_finder.run(dataloader_plot)
+    with lr_finder.attach(dummy_engine_mnist, mnist_to_save) as trainer_with_finder:
+        trainer_with_finder.run(mnist_dataloader)
 
-    # Avoid RuntimeError
```

Comment:
Can you also update `test_lr_suggestion` and remove : 
```python
    # Avoid RuntimeError
    if torch.tensor(lr_finder._history["loss"]).argmin() < 2:
        lr_finder._history["loss"].insert(0, 10)
        lr_finder._history["loss"].insert(0, 100)
        lr_finder._history["lr"].insert(0, 10)
        lr_finder._history["lr"].insert(0, 100)

    assert 1e-4 <= lr_finder.lr_suggestion() <= 10
```
from there ?

Reply:
I thought about that, but If we did that we will have to MNIST dataset, and if we did that, there will be no difference between `test_lr_suggestion` and `test_mnist_lr_suggestion`, so we can leave `test_lr_suggestion` as it is or we can delete it and rename `test_mnist_lr_suggestion` to `test_lr_suggestion`, and consider it the learning rate test function, what do you think?

Reply:
Let's do the following: let's manually create loss/lr history without calling
```python
    with lr_finder.attach(dummy_engine, to_save) as trainer_with_finder:
        trainer_with_finder.run(dataloader)
```
and check lr_suggestion on this synthetic data.
Another comment:
```python
assert 1e-4 <= lr_finder.lr_suggestion() <= 10
```
this assertion should be stricter, internal [1e-4, 10] is too large.